x/concentrated-liquidity/internal/math/tick_test.go

```diff
@@ -1,68 +1,242 @@
+			tickIndex:          sdk.NewInt(100),
+			exponentAtPriceOne: types.ExponentAtPriceOneMax.Add(sdk.OneInt()),
+			expectedError:      types.ExponentAtPriceOneError{ProvidedExponentAtPriceOne: types.ExponentAtPriceOneMax.Add(sdk.OneInt()), PrecisionValueAtPriceOneMin: types.ExponentAtPriceOneMin, PrecisionValueAtPriceOneMax: types.ExponentAtPriceOneMax},
+		},
+		"random": {
+			tickIndex:          sdk.NewInt(-9111000000),
+			exponentAtPriceOne: sdk.NewInt(-8),
+			expectedPrice:      sdk.MustNewDecFromStr("0.000000000088900000"),
 		},
 	}
 
 	for name, tc := range testCases {
 		tc := tc
 
 		suite.Run(name, func() {
-			sqrtPrice, err := math.TickToSqrtPrice(tc.tickIndex)
+			sqrtPrice, err := math.TickToPrice(tc.tickIndex, tc.exponentAtPriceOne)
+			if tc.expectedError != nil {
+				suite.Require().Error(err)
+				suite.Require().Equal(tc.expectedError.Error(), err.Error())
+				return
+			}
 			suite.Require().NoError(err)
-			suite.Require().Equal(tc.sqrtPriceExpected, sqrtPrice.String())
+			suite.Require().Equal(tc.expectedPrice.String(), sqrtPrice.String())
 
 		})
 	}
 }
 
 func (suite *ConcentratedMathTestSuite) TestPriceToTick() {
-	testCases := []struct {
-		name         string
-		price        sdk.Dec
-		tickExpected string
+	testCases := map[string]struct {
+		price              sdk.Dec
+		exponentAtPriceOne sdk.Int
+		tickExpected       string
+		expectedError      error
 	}{
-		{
-			"happy path",
-			sdk.NewDec(5000),
-			"85176",
+		"50000 to tick with -4 k at price one": {
+			price:              sdk.NewDec(50000),
+			exponentAtPriceOne: sdk.NewInt(-4),
+			tickExpected:       "400000",
+		},
+		"5.01 to tick with -2 k at price one": {
+			price:              sdk.MustNewDecFromStr("5.01"),
+			exponentAtPriceOne: sdk.NewInt(-2),
+			tickExpected:       "401",
+		},
+		"50000.01 to tick with -6 k at price one": {
+			price:              sdk.MustNewDecFromStr("50000.01"),
+			exponentAtPriceOne: sdk.NewInt(-6),
+			tickExpected:       "40000001",
+		},
+		"0.090000889 to tick with -8 k at price one": {
+			price:              sdk.MustNewDecFromStr("0.090000889"),
+			exponentAtPriceOne: sdk.NewInt(-8),
+			tickExpected:       "-999991110",
+		},
+		"0.9998 to tick with -4 k at price one": {
+			price:              sdk.MustNewDecFromStr("0.9998"),
+			exponentAtPriceOne: sdk.NewInt(-4),
+			tickExpected:       "-20",
+		},
+		"53030.10 to tick with -5 k at price one": {
+			price:              sdk.MustNewDecFromStr("53030.1"),
+			exponentAtPriceOne: sdk.NewInt(-5),
+			tickExpected:       "4030301",
+		},
+		"max spot price and minimum exponentAtPriceOne": {
+			price:              MaxSpotPrice,
+			exponentAtPriceOne: sdk.NewInt(-1),
+			tickExpected:       "3420",
+		},
+		"error: price must be positive": {
+			price:              sdk.NewDec(-1),
+			exponentAtPriceOne: sdk.NewInt(-6),
+			expectedError:      fmt.Errorf("price must be greater than zero"),
+		},
+		"error: resulting tickIndex too large": {
```

Comment:
Please also add "error: resulting tickIndex too small"

Reply:
I just added a check to check price is within the allowed spot price bounds before this check. I do not know how to trigger either tickIndex too large or tickIndex too small error, and it feels like it is just a defense in depth at this point